src/wallet/rpcdump.cpp

```diff
@@ -1482,3 +1482,263 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
+                        "\"requests\""},
+                },
+                RPCResult{
+            "\nResponse is an array with the same size as the input that has the execution result :\n"
+            "  [{\"success\": true}, {\"success\": true, \"warnings\": [\"Ignoring irrelevant private key\"]}, {\"success\": false, \"error\": {\"code\": -1, \"message\": \"Internal Server Error\"}}, ...]\n"
+                },
+                RPCExamples{
+                    HelpExampleCli("importdescriptors", "'[{ \"desc\": \"<my descriptor>\", \"timestamp\":1455191478, \"address_type\": \"legacy\", \"internal\": true }, "
+                                          "{ \"desc\": \"<my desccriptor 2>\", \"label\": \"example 2\", \"timestamp\": 1455191480, \"address_type\": \"legacy\" }]'") +
+                    HelpExampleCli("importdescriptors", "'[{ \"desc\": \"<my descriptor>\", \"timestamp\":1455191478, \"address_type\": \"bech32\", \"active\": true, \"range\": [0,100], \"label\": \"<my bech32 wallet>\" }]'")
+                },
+            }.ToString()
+        );
+
+    //  Make sure wallet is a descriptor wallet
+    if (!pwallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "importdescriptors is not available for non-descriptor wallets");
+    }
+
+    RPCTypeCheck(main_request.params, {UniValue::VARR, UniValue::VOBJ});
+
+    WalletRescanReserver reserver(pwallet);
+    if (!reserver.reserve()) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Wallet is currently rescanning. Abort existing rescan or wait.");
+    }
+
+    const UniValue& requests = main_request.params[0];
+    const int64_t minimum_timestamp = 1;
+    int64_t now = 0;
+    int64_t lowest_timestamp = 0;
+    bool rescan = false;
+    UniValue response(UniValue::VARR);
+    {
+        auto locked_chain = pwallet->chain().lock();
+        LOCK(pwallet->cs_wallet);
+        EnsureWalletIsUnlocked(pwallet);
+
+        const Optional<int> tip_height = locked_chain->getHeight();
+        if (tip_height) {
+            now = locked_chain->getBlockMedianTimePast(*tip_height);
+            // initialize lowest timestamp to chain tip
+            lowest_timestamp = locked_chain->getBlockTime(*tip_height);
+        }
+
+        // Get all timestamps and extract the lowest timestamp
+        for (const UniValue& request : requests.getValues()) {
+            // This throws an error if "timestamp" doesn't exist
+            const int64_t timestamp = std::max(GetImportTimestamp(request, now), minimum_timestamp);
+            const UniValue result = ProcessDescriptorImport(pwallet, request, timestamp);
+            response.push_back(result);
+
+            if (lowest_timestamp > timestamp ) {
+                lowest_timestamp = timestamp;
+            }
+
+            // If we know the chain tip, and at least one request was successful then allow rescan
+            // TODO: should we throw an error if we couldn't find the tip?
```

Comment:
I'm pretty sure no tip just means that the node hasn't started syncing yet and is still at the genesis block. I think it's fine to ignore if `tip_height` doesn't exist. It is only being used to get the "now" time so it doesn't matter if we ignore it.

Reply:
Cool. Yeah the worst case here is not that bad so probably not a big deal.

Reply:
Also `impormulti` doesn't handle that case at all, and I think it is safe to follow it.